### PR TITLE
Add XSL transform for article listicles

### PR DESCRIPTION
**Pull request by i-like-robots**
This is a bit hacky while we combine the use of Cheerio + XSLT.

## Before

screen shot 2015-06-25 at 17 06 48

## After

screen shot 2015-06-25 at 17 08 56

Related: https://github.com/Financial-Times/next-article/issues/585

### CODE REVIEW

File: server/transforms/listicle-xslt.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+var denodeify = require('denodeify');
+var libxslt = require('libxslt');
+
+module.exports = function bigReadTransform(articles) {
```
Comment (ironsidevsquincy) [inline server/transforms/listicle-xslt.js]: just pass in the single article that needs transforming

File: server/controllers/article.js [hunk 2]
```diff
@@ -42,6 +43,13 @@ module.exports = function(req, res, next) {
 
 	Promise.all([articleV1Promise, articleV2Promise])
 		.then(function(articles) {
+			if (extractUuid(articles[1].id) === '54fba5c4-e2d6-11e4-aa1d-00144feab7de') {
+				return listicleXSLT(articles);
```
Comment (ironsidevsquincy) [inline server/controllers/article.js]: as below,

```
articles[1] = listicleXSLT(articles[1]);
```

File: server/stylesheets/article.xsl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+							</header>
+							<p class="breakout__item-description">
+								<xsl:value-of select="normalize-space()" />
+							</p>
+						</div>
+					</li>
+				</xsl:for-each>
+			</ul>
+		</xsl:if>
+	</xsl:template>
```
Comment (ironsidevsquincy) [inline server/stylesheets/article.xsl]: jeez, yeah, I really need to bone up xslt

File: server/transforms/listicle-xslt.js (hunk 1 above)
Comment (ironsidevsquincy) [inline server/transforms/listicle-xslt.js]: or, better, just pass in the xml that needs transforming

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/11:03-2140/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/11:03-2140/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/11:14-2142/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/11:14-2142/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/11:50-2144/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/11:50-2144/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/13:59-2146/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/13:59-2146/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/14:22-2148/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-24/14:22-2148/successes/index.html)

**Comment by ironsidevsquincy** [comment]
Couple of these diffs look odd, but there definitely is a difference with the promo box's headers

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-25/13:29-2189/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-06-25/13:29-2189/successes/index.html)

**Comment by ironsidevsquincy** [comment]
Is it behind a flag?

File: server/transforms/article-xslt.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+function wrap(article) {
+	return '<root>' + article + '</root>';
+}
+
+function unwrap(article) {
+	return article.replace(/<\/?root>/g, '');
+}
+
+module.exports = function bigReadTransform(article) {
+	var parseFile = denodeify(libxslt.parseFile);
```
Comment (commuterjoy) [inline server/transforms/article-xslt.js]: Do you want to do this once per application, or once per transform. I think the former?

**Comment by i-like-robots** [comment]
@ironsidevsquincy It can be! https://github.com/Financial-Times/next-feature-flags-api/pull/195

**Comment by ironsidevsquincy** [comment]
:+1:

**Pull request merged**